### ISSUE TITLE
Alternate models experimentation (attention)

### ISSUE THREAD

**Issue by felix-e-h-p** [openclimatefix/pvnet-summation#47]
Current summation model and input structure being flattened PVNet outputs per horizon, think that boosting models could be a good alternative/candidate to benchmark against.

Considering sample count, definitely well within their effective range and require less tuning. Could prove to be faster overall, and definitely interested to see accuracy etc.

Will aim to implement two new model classes as alternatives to HorizonDenseModel: LightGBM and XGB - one regressor per output quantile that fits in a single pass at end of initial epoch and stops.

Potential changes (to be updated sporadically):

- Add: pvnet_summation/models/lgbm_model.py
- Add: pvnet_summation/models/xgb_model.py
- Edit: pvnet_summation/training/lightning_module.py (boosting branch in training_step and on_train_epoch_end)

Compare val MAE and quantile calibration of LightGBM and XGBoost against current HorizonDenseModel baseline (maybe with NL runs?).

**Comment by felix-e-h-p** [comment]
LightGBM and XGBoost benchmarked against HorizonDenseModel and both significantly underperformed even with extensive tuning etc.

Implemented LocationAttentionModel with:

- Multi-head self-attention over location dimension at each horizon step
- Learned GSP identity embeddings so attention can understand locations
- Cross-horizon attention block to captures dependencies across forecast steps
- Learned horizon positional embeddings

Changes for this - will discard boosting related files etc:

- Add: pvnet_summation/models/attention_model.py
- Edit: pvnet_summation/models/__init__.py
- Add: configs/model/model_attention.yaml

Gonna confirm best variant RE attention across multiple seeds before pushing to branch - will be in a full write up.

**Comment by felix-e-h-p** [comment]
https://github.com/openclimatefix/pvnet-summation/compare/alternate-models

**Comment by felix-e-h-p** [comment]
<img width="1485" height="733" alt="Image" src="https://github.com/user-attachments/assets/69d6e27f-0a70-4b5d-8abc-1611bb53ef6d" />
<img width="1485" height="733" alt="Image" src="https://github.com/user-attachments/assets/3c72ffe6-a0d9-49a7-b618-418a99a0b280" />

**Comment by felix-e-h-p** [comment]
Results from attention model experiments on NL solar summation, with the 3 MO features removed.

Implemented `LocationAttentionModel` with three different varients: location only attention (v0), tuned with learned GSP embeddings and stacked layers (v1), and full cross-horizon attention (v2).

| Model | MAE/val | Δ vs baseline | quantile_loss/val | Δ vs baseline |
|---|---|---|---|---|
| Summation baseline | 0.01791 | — | 0.00988 | — |
| Attention v0 | 0.01804 | -0.7% | 0.01008 | -2.0% |
| Attention v1 (tuned) | 0.01581 | +11.7% | 0.00875 | +11.4% |
| Attention v2 (cross-horizon) | 0.00969 | **+45.9%** | 0.00525 | **+46.8%** |

v2 adds a second attention block across the forecast horizon steps with learned positional embeddings. The approx 47% MAE improvement over the baseline is pretty strong! Ran three other different seeded runs with v2, relatively similar metrics.
